platform/cog-platform-fdo.c

```diff
@@ -1751,6 +1892,107 @@ destroy_window (void)
 #endif
 }
 
+static void
+create_popup (WebKitOptionMenu *option_menu)
```

Comment:
There is quite some churn going on in this function, and I wonder whether it might be possible to move it over to `cog-popup-menu-fdo.c` somehow. Anyway, we can leave it here for now and take care of that later on for the `devel/multiple-views` branch, which has some abstraction utility code for handling Wayland “windows”—no need to do that now.

Reply:
Not sure pushing the Wayland shell logic into `cog-popup-menu-fdo.c` is beneficial. I would rather (in the future, pos-multiview breakage) prefer that the specific shell logic is all combined in separate implementation files.